nucypher/characters/lawful.py

```diff
@@ -808,27 +792,37 @@ def _reencrypt(self,
         self._completed_work_orders.save_work_order(work_order, as_replete=retain_cfrags)
 
         the_airing_of_grievances = []
+        verified_cfrags = []
```

Comment:
In the interest in keeping this ever-growing method tidy (it's expanded in #2687) what do you think about making this block into a method?  It also might improve testability.

Reply:
For this PR I wanted to limit the changes to Umbral compatibility as much as possible, but I did plan to go through the whole retrieval flow in a separate PR (which will include the changes for #259). The problem here is that we still need to check the cfrag somewhere within `_reencrypt()` so that we could report a fail and fall back on the next Ursula. I'm sure it can be improved, of course.